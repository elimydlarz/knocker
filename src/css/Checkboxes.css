.Checkboxes {
  margin-bottom: 15px;
}

.checkbox-heading {
  margin-bottom: 1px;
}

.checkbox-label {
  font-weight: 500;
}

input[type="checkbox"] {
  margin: 0 7px 0 0;
}
