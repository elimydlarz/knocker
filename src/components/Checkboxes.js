import React, { Component } from 'react';
import '../css/Checkboxes.css';

export default class Checkboxes extends Component {
  render() { return (
    <div className='Checkboxes'>
      <label>
        <div className='checkbox-heading'>
          {this.props.label}
        </div>
        {this.props.options.map(option =>
          <label className='checkbox-label' key={option}>
            <input name={this.props.name} type='checkbox'
              value={option}
              checked={this.props.checkedOptions.includes(option)}
              onClick={this.props.changeHandler} />
            {option}
          </label>
        )}
      </label>
    </div>
  ); }
}
