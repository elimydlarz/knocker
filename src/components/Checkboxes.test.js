import React from 'react';
import renderer from 'react-test-renderer';
import Checkboxes from './Checkboxes';

describe('Checkboxes', () => {
  it('does not explode', () => {
    const component = renderer.create(
      <Checkboxes
        name='COOL NAME'
        checkedOptions={['COOL OPTION']}
        changeHandler={() => {}}
        label='COOL LABEL'
        options={[
          'COOL OPTION',
          'UNCOOL OPTION',
        ]}
      />
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
