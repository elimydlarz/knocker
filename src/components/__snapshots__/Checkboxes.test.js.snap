exports[`Checkboxes does not explode 1`] = `
<div
  className="Checkboxes">
  <label>
    <div
      className="checkbox-heading">
      COOL LABEL
    </div>
    <label
      className="checkbox-label">
      <input
        checked={true}
        name="COOL NAME"
        onClick={[Function changeHandler]}
        type="checkbox"
        value="COOL OPTION" />
      COOL OPTION
    </label>
    <label
      className="checkbox-label">
      <input
        checked={false}
        name="COOL NAME"
        onClick={[Function changeHandler]}
        type="checkbox"
        value="UNCOOL OPTION" />
      UNCOOL OPTION
    </label>
  </label>
</div>
`;
